test/test.getch.ts

```diff
@@ -250,14 +250,181 @@ describe('🥁 configuration options', () => {
+      const scope = nock(url).get('/').reply(200, body);
+      const res = await request({url});
+      scope.done();
+      assert.deepStrictEqual(res.data, body);
+      assert.ok(res.config.agent instanceof HttpsProxyAgent);
+    });
+
+    it('should not proxy if no_proxy env variable starts with a dot', async () => {
+      const url = 'https://domain.example.com';
+      sandbox.stub(process, 'env').value({
+        https_proxy: 'https://fake.proxy',
+        no_proxy: '.example.com',
+      });
+      const body = {hello: '🌎'};
+      const scope = nock(url).get('/').reply(200, body);
+      const res = await request({url});
+      scope.done();
+      assert.deepStrictEqual(res.data, body);
+      assert.strictEqual(res.config.agent, undefined);
+    });
+
+    it('should proxy if proxy env variable starts with a dot', async () => {
+      const url = 'https://domain.example.com';
+      sandbox.stub(process, 'env').value({
+        https_proxy: '.example.com',
+      });
+      const body = {hello: '🌎'};
+      const scope = nock(url).get('/').reply(200, body);
+      const res = await request({url});
+      scope.done();
+      assert.deepStrictEqual(res.data, body);
+      assert.ok(res.config.agent instanceof HttpsProxyAgent);
+    });
+
+    it('should allow comma-separated lists for no_proxy env variables', async () => {
+      const url = 'https://api.google.com';
+      sandbox.stub(process, 'env').value({
+        https_proxy: 'https://fake.proxy',
+        no_proxy: 'example.com,*.google.com,hello.com',
+      });
+      const body = {hello: '🌎'};
+      const scope = nock(url).get('/').reply(200, body);
+      const res = await request({url});
+      scope.done();
+      assert.deepStrictEqual(res.data, body);
+      assert.strictEqual(res.config.agent, undefined);
+    });
+
+    it('should allow comma-separated lists for proxy env variables', async () => {
+      const url = 'https://api.google.com';
+      sandbox.stub(process, 'env').value({
+        https_proxy: 'example.com,*.google.com,hello.com',
+      });
+      const body = {hello: '🌎'};
+      const scope = nock(url).get('/').reply(200, body);
+      const res = await request({url});
+      scope.done();
+      assert.deepStrictEqual(res.data, body);
+      assert.ok(res.config.agent instanceof HttpsProxyAgent);
+    });
+
+    it('should let no_proxy take precedence, if url is in both proxy and no_proxy env variables', async () => {
+      const url = 'https://example.com';
+      sandbox.stub(process, 'env').value({
+        https_proxy: 'example.com',
+        no_proxy: 'example.com',
+      });
+      const body = {hello: '🌎'};
+      const scope = nock(url).get('/').reply(200, body);
+      const res = await request({url});
+      scope.done();
+      assert.deepStrictEqual(res.data, body);
+      assert.strictEqual(res.config.agent, undefined);
+    });
+
+    it('should proxy if url is an exact match of proxy, but not of no_proxy', async () => {
```

Comment:
we should also update this test to reflect the fact that there's no correlation between the URL and the `https_proxy` variable.

It's `no_proxy` that matches a list of URLs.